torch/_refs/__init__.py

```diff
@@ -5744,24 +5744,78 @@ def log_normal(self, mean=1, std=2, generator=None):
 
 
 # TODO: add support for functionalization aten.normal_functional
+# NOTE: we cannot use the out wrapper because for the overload
+# normal.float_float_out we need to get the device and dtype from the out parameter
 @register_decomposition(aten.normal)
-@out_wrapper()
 @elementwise_type_promotion_wrapper(
-    type_promoting_args=("self",),
+    type_promoting_args=(
+        "mean",
+        "std",
+    ),
     type_promotion_kind=ELEMENTWISE_TYPE_PROMOTION_KIND.DEFAULT,
 )
-def normal(self, mean=0, std=1, generator=None):
+def normal(
+    mean=0,
+    std=1,
+    shape=None,
+    *,
+    generator=None,
+    out=None,
+    dtype=None,
+    layout=None,
+    device=None,
+    pin_memory=None,
+):
     assert generator is None
-    torch._check(std >= 0, lambda: f"normal expects std >= 0.0, but found std {std}")
+    assert layout is None
+
+    torch._check(
+        isinstance(mean, TensorLike)
+        or isinstance(std, TensorLike)
+        or shape is not None,
+        lambda: "normal expects that either mean or std is a tensor, or shape is defined",
+    )
```

Comment:
This is not right yet. Shape is only accepted when `mean` and `std` are both scalars.

Reply:
I'd argue that adding that restriction makes the code more complicated for very little benefit. However I'm not the maintainer so I'll make the change.

Reply:
Sorry still not fixed

Reply:
I'd generally agree with you, but in this case, if you don't put that distinction, you would have an ambiguity where it's not clear what shape would you choose if you pass both `shape` and a tensor.

Reply:
Yep that's a good point

Reply:
I guess you could broadcast in that case - but we're definitely moving away from the definition of the op